R/format_GRO.R

```diff
@@ -0,0 +1,418 @@
+  Capture_data <- create_capture_GRO(gro_data)
+
+  #### INDIVIDUAL DATA
+  message("Compiling individual information...")
+  Individual_data <- create_individual_GRO(Capture_data)
+
+  #### LOCATION DATA
+  message("Compiling location information...")
+  Location_data <- create_location_GRO(gro_data)
+
+  time <- difftime(Sys.time(), start_time, units = "sec")
+
+  message(paste0("All tables generated in ", round(time, 2), " seconds"))
+
+  #### EXPORT DATA
+
+  if(output_type == "csv"){
+
+    message("Saving .csv files...")
+
+    utils::write.csv(x = Brood_data, file = paste0(path, "\\Brood_data_GRO.csv"), row.names = F)
+
+    utils::write.csv(x = Capture_data, file = paste0(path, "\\Capture_data_GRO.csv"), row.names = F)
+
+    utils::write.csv(x = Individual_data, file = paste0(path, "\\Individual_data_GRO.csv"), row.names = F)
+
+    utils::write.csv(x = Location_data, file = paste0(path, "\\Location_data_GRO.csv"), row.names = F)
+
+    invisible(NULL)
+
+  }
+
+  if(output_type == "R"){
+
+    message("Returning R objects...")
+
+    return(list(Brood_data = Brood_data,
+                Capture_data = Capture_data,
+                Individual_data = Individual_data,
+                Location_data = Location_data))
+
+  }
+
+}
+
+#### --------------------------------------------------------------------------~
+#### FUNCTIONS
+#### --------------------------------------------------------------------------~
+
+
+#' Create brood data table for great tits and blue tits in Grobla, Poland.
+#'
+#' @param gro_data Data frame of modified primary data from Grobla, Poland.
+#'
+#' @return A data frame.
+
+create_brood_GRO <- function(gro_data) {
+
+  ## Get brood data
+  Brood_data <- gro_data %>%
+    dplyr::filter(!is.na(.data$LocationID)) %>%
+
+    ## Summarize brood information for each nest
+    dplyr::group_by(.data$BreedingSeason, .data$PopID, .data$LocationID, .data$LayDate_observed) %>%
+
+    dplyr::mutate(FemaleID = .data$FemaleID,
+                  MaleID = .data$MaleID)  %>%
+
+    ## Keep only necessary columns
+    dplyr::select(dplyr::contains(names(brood_data_template))) %>%
+
+    ## Add missing columns
+    dplyr::bind_cols(brood_data_template[,!(names(brood_data_template) %in% names(.))]) %>%
+
+    ## Reorder columns
+    dplyr::select(names(brood_data_template)) %>%
+
+    ## Remove any NAs from essential columns
+    dplyr::filter(!is.na(.data$PopID),
+                  !is.na(.data$BreedingSeason),
+                  !is.na(.data$Species)) %>%
+
+    ## Calculate clutch type
+    dplyr::arrange(.data$PopID, .data$BreedingSeason, .data$Species, .data$FemaleID, .data$LayDate_observed) %>%
+    dplyr::ungroup() %>%
+    dplyr::mutate(ClutchType_calculated = calc_clutchtype(data =. , protocol_version = "1.1", na.rm = FALSE)) %>%
+
+    ## Calculating BroodID if Species is known
+    dplyr::mutate(BroodID = dplyr::case_when(!is.na(Species) ~ paste(.data$PopID, dplyr::row_number(), sep ="-")))
```

Comment:
Good point. There are currently no cases where the Species is unknown, so I wasn't thinking about filtering them out, but there certainly could be and then I think it would make sense to remove them from the Brood data since Species information is required. I've added a filter to remove any NAs in BroodID